azurelinuxagent/common/rdma.py

```diff
@@ -29,19 +29,16 @@
 import azurelinuxagent.common.utils.shellutil as shellutil
 from azurelinuxagent.common.utils.textutil import parse_doc, find, getattrib
 
-
-from azurelinuxagent.common.protocol.wire import SHARED_CONF_FILE_NAME
-
 dapl_config_paths = [
     '/etc/dat.conf',
     '/etc/rdma/dat.conf',
     '/usr/local/etc/dat.conf'
 ]
 
-def setup_rdma_device(nd_version):
+
+def setup_rdma_device(nd_version, shared_conf):
     logger.verbose("Parsing SharedConfig XML contents for RDMA details")
-    xml_doc = parse_doc(
-        fileutil.read_file(os.path.join(conf.get_lib_dir(), SHARED_CONF_FILE_NAME)))
+    xml_doc = parse_doc(shared_conf.xml_text)
```

Comment:
I did not notice this parsing code before... can we move it to wire.py (SharedConfig)

Reply:
Sure, I'll review it and move it there